src/Asset/EntrypointLookup.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\WebpackEncoreBundle\Asset;
+
+/**
+ * Returns the CSS or JavaScript files needed for a Webpack entry.
+ *
+ * This reads a JSON file with the format of Webpack Encore's entrypoints.json file.
+ * @final
+ */
+class EntrypointLookup
+{
+    private $entrypointJsonPath;
+
+    private $entriesData;
+
+    private $returnedFiles = [];
+
+    public function __construct(string $entrypointJsonPath)
+    {
+        $this->entrypointJsonPath = $entrypointJsonPath;
+    }
+
+    public function getJavaScriptFiles(string $entryName): array
+    {
+        return $this->getEntryFiles($entryName, 'js');
+    }
+
+    public function getCssFiles(string $entryName): array
+    {
+        return $this->getEntryFiles($entryName, 'css');
+    }
+
+    private function getEntryFiles(string $entryName, string $key): array
+    {
+        $this->validateEntryName($entryName);
+        $entriesData = $this->getEntriesData();
+        $entryData = $entriesData[$entryName];
+
+        if (!isset($entryData[$key])) {
+            throw new \InvalidArgumentException(sprintf('Could not find "%s" key for the "%s" entry.', $key, $entryName));
+        }
+
+        // make sure to not return the same file multiple times
+        $entryFiles = $entryData[$key];
+        $newFiles = array_values(array_diff($entryFiles, $this->returnedFiles));
+        $this->returnedFiles = array_merge($this->returnedFiles, $newFiles);
+
+        return $newFiles;
+    }
+
+    private function validateEntryName(string $entryName)
+    {
+        $entriesData = $this->getEntriesData();
+        if (!isset($entriesData[$entryName])) {
+            $withoutExtension = substr($entryName, 0, strrpos($entryName, '.'));
```

Comment:
what happens if there is no dot in `$entryName` ?

Reply:
It seems to be ok - I have some test cases to make sure we fail with the correct exception for simply wrong entry names - e.g. `foo` as well as wrong entry names just because you added the extension - `foo.js`